kyo-core/shared/src/main/scala/kyo/System.scala

```diff
@@ -184,32 +187,32 @@ object System:
     /** Companion object for Parser, containing default implementations. */
     object Parser:
         given Parser[Nothing, String]                    = v => Result.success(v)
-        given Parser[NumberFormatException, Int]         = v => Result.catching(v.toInt)
-        given Parser[NumberFormatException, Long]        = v => Result.catching(v.toLong)
-        given Parser[NumberFormatException, Float]       = v => Result.catching(v.toFloat)
-        given Parser[NumberFormatException, Double]      = v => Result.catching(v.toDouble)
-        given Parser[IllegalArgumentException, Boolean]  = v => Result.catching(v.toBoolean)
-        given Parser[NumberFormatException, Byte]        = v => Result.catching(v.toByte)
-        given Parser[NumberFormatException, Short]       = v => Result.catching(v.toShort)
+        given Parser[NumberFormatException, Int]         = v => Result.catching[NumberFormatException](v.toInt)
```

Comment:
After https://github.com/getkyo/kyo/pull/708/commits/076a93f98653672cdfafd214bdc5aeb061ffa8a7, I wasn't sure why this class still needed changes for the tests to pass. It turns out that the compiler was inferring `ClassTag[Nothing]`, which incorrectly becomes `java.lang.Object` at runtime and transforms any exceptions to `Fail`. `SafeClassTag` fixes this issue but we now need the explicit type parameter to avoid a panic.